pymc3/step_methods/metropolis.py

```diff
@@ -400,9 +400,14 @@ def __init__(self, vars, proposal='uniform', order='random', model=None):
                 k = draw_values([distr.k])[0]
             elif isinstance(distr, pm.Bernoulli) or (v.dtype in pm.bool_types):
                 k = 2
+            elif isinstance(distr, pm.DiscreteUniform):
+                k = draw_values([distr.upper])[0]
+                if draw_values([distr.lower])[0] != 0:
```

Comment:
Draw values of `lower` could return an array, and not a scalar. You should take this into account and wrap the equality comparison in an `np.all`

Reply:
@lucianopaz I have amended the change and also added PRs for examples and Release Notes. This is my first PR, so I appreciate your assistance with this.

Reply:
@lucianopaz I have amended the test code to improve coverage. The travis tests seem to have been affected by matplotlib issue #3443.